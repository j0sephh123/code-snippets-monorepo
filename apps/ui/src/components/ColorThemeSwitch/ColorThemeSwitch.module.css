/* 
  Code copied from this example:

  https://ui.mantine.dev/component/action-toggle/
  https://mantine.dev/overview/#postcss-preset 
*/

.icon {
  width: rem(22px);
  height: rem(22px);
}

.dark {
  @mixin dark {
    display: none;
  }

  @mixin light {
    display: block;
  }
}

.light {
  @mixin light {
    display: none;
  }

  @mixin dark {
    display: block;
  }
}
