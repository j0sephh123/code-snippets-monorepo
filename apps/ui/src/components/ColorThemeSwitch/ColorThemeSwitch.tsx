import cx from 'clsx';
import {
  ActionIcon,
  useMantineColorScheme,
  useComputedColorScheme,
  Group,
} from '@mantine/core';
import { IconSun, IconMoon } from '@tabler/icons-react';
import classes from './ColorThemeSwitch.module.css';

// Reused from this example:
// https://ui.mantine.dev/component/action-toggle/
export default function ColorThemeSwitch() {
  const { setColorScheme } = useMantineColorScheme();
  const computedColorScheme = useComputedColorScheme('dark', {
    getInitialValueInEffect: true,
  });

  return (
    <Group justify="center">
      <ActionIcon
        onClick={() =>
          setColorScheme(computedColorScheme === 'light' ? 'dark' : 'light')
        }
        variant="default"
        size="xl"
        aria-label="Toggle color scheme"
      >
        <IconSun className={cx(classes.icon, classes.light)} stroke={1.5} />
        <IconMoon className={cx(classes.icon, classes.dark)} stroke={1.5} />
      </ActionIcon>
    </Group>
  );
}
