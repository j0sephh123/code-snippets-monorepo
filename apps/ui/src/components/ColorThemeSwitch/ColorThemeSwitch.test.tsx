import userEvent from '@testing-library/user-event';
import { render, screen } from '@testing-library/react';
import ColorThemeSwitch from './ColorThemeSwitch';
import { MantineProvider, createTheme } from '@mantine/core';

const setColorSchemeSpy = vi.hoisted(vi.fn);

vi.mock('@mantine/core', async () => {
  const mantineActual = await vi.importActual('@mantine/core');

  return {
    ...(mantineActual as any),
    useMantineColorScheme: () => ({ setColorScheme: setColorSchemeSpy }),
  };
});

describe('components > ColorThemeSwitch', () => {
  it('renders', async () => {
    const user = userEvent.setup();
    const { asFragment } = render(
      <MantineProvider defaultColorScheme="dark" theme={createTheme({})}>
        <ColorThemeSwitch />
      </MantineProvider>
    );
    const btn = screen.getByRole('button');

    await user.click(btn);

    expect(setColorSchemeSpy).toHaveBeenCalledWith('light');
    expect(asFragment()).toMatchSnapshot();
  });
});
