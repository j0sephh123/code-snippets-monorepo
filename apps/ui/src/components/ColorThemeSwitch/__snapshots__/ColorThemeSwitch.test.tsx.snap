// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components > ColorThemeSwitch > renders 1`] = `
<DocumentFragment>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="m-4081bf90 mantine-Group-root"
    style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: center; --group-wrap: wrap;"
  >
    <button
      aria-describedby=":r2:"
      aria-label="Toggle color scheme"
      class="mantine-focus-auto mantine-active m-8d3f4000 mantine-ActionIcon-root m-87cf2631 mantine-UnstyledButton-root"
      data-size="xl"
      data-variant="default"
      style="--ai-size: var(--ai-size-xl); --ai-bg: var(--mantine-color-default); --ai-hover: var(--mantine-color-default-hover); --ai-color: var(--mantine-color-default-color); --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid var(--mantine-color-default-border);"
      type="button"
    >
      <svg
        class="_icon_1eb3e8 _light_1eb3e8"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 12m-4 0a4 4 0 1 0 8 0a4 4 0 1 0 -8 0"
        />
        <path
          d="M3 12h1m8 -9v1m8 8h1m-9 8v1m-6.4 -15.4l.7 .7m12.1 -.7l-.7 .7m0 11.4l.7 .7m-12.1 -.7l-.7 .7"
        />
      </svg>
      <svg
        class="_icon_1eb3e8 _dark_1eb3e8"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
